### PR TITLE
Add timestamps to XUnitLogChecker output

### PR DESCRIPTION
**Pull request by markples**
The main motivation here is to give a quick notification whether a timeout was due to taking a long time or coincidentally hitting some external limit at that point.  I think the below is due to the `watchdog 119` hitting, and since 9:31:47 isn't long after 9:31:44, then the thread07 test is probably stuck, but I'm never sure that I'm reading it correctly.

```
/root/helix/work/correlation/watchdog 119 /root/helix/work/correlation/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false -p System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization=true threading_group2.dll ''
09:31:44.737 Running test: baseservices/threading/generics/Monitor/EnterExit01/EnterExit01.dll
Test Passed
09:31:44.875 Passed test: baseservices/threading/generics/Monitor/EnterExit01/EnterExit01.dll
...
09:31:47.302 Running test: baseservices/threading/generics/TimerCallback/thread07_TimerCallback/thread07_TimerCallback.dll
Test Passed
09:31:47.510 Passed test: baseservices/threading/generics/TimerCallback/thread07_TimerCallback/thread07_TimerCallback.dll
09:31:47.512 Running test: baseservices/threading/generics/WaitCallback/thread07_WaitCallback/thread07_WaitCallback.dll
Child process took too long. Timed out... Exiting...
App Exit Code: 110
Expected: 100
Actual: 110
END EXECUTION - FAILED
+ test_exit_code=1
+ dotnet /root/helix/work/correlation/XUnitLogChecker/XUnitLogChecker.dll baseservices/threading/threading_group2 threading_group2 /home/helixbot/dotnetbuild/dumps
[XUnitLogChecker]: Item 'threading_group2' did not finish running. Checking and fixing the log...
[XUnitLogChecker]: XUnit log file has been fixed!

28/95 tests run.
* 28 tests passed.
* 0 tests failed.
* 0 tests skipped.

[XUnitLogChecker]: Checking for dumps...
```

Here's a snippet from a successful test run for this PR. If the "passed test" line was missing, we would still easily be able to bound the test time under 3 seconds.

```
18:44:42.835 Running test: JIT/jit64/hfa/main/testC/hfa_nd1C_r/hfa_nd1C_r.cmd
<snip>
18:44:45.041 Passed test: JIT/jit64/hfa/main/testC/hfa_nd1C_r/hfa_nd1C_r.cmd
App Exit Code: 100
Expected: 100
Actual: 100
END EXECUTION - PASSED
PASSED
[XUnitLogChecker]: 18:44:45.25: Item 'jit64_1' did complete successfully!

### CODE REVIEW

**Comment by markples** [comment]
Failures are marked known plus #85993

**Comment by markples** [comment]
@ivdiazsa Does adding these timestamps seem reasonable to you?

File: src/tests/Common/XUnitLogChecker/XUnitLogChecker.cs [hunk 1]
```diff
@@ -154,19 +153,25 @@ static int Main(string[] args)
...
         // knowing what transpired here.
         File.Move(tempLogPath, finalLogPath);
-        Console.WriteLine("[XUnitLogChecker]: Finished!");
+        WriteLineTimestamp("Finished!");
         return SUCCESS;
     }
 
+    static void WriteLineTimestamp(string message)
+    {
+        Console.Write($"[XUnitLogChecker]: {System.DateTime.Now:HH:mm:ss.ff}: ");
```
Comment (danmoseley) [inline src/tests/Common/XUnitLogChecker/XUnitLogChecker.cs:171]: Is any concurrent console output possible? I assume not but if it is you might want to make this a single write to the console not two.

**Review by ivdiazsa — APPROVED** [review]
I think adding the timestamps will definitely prove to be a helpful change. Thanks for doing this work, Mark! And thanks for keeping the format to avoid super long lines of code. LGTM!

I just kicked off an outerloop CI run to be safe. I've had experiences of those breaking in the most unexpected ways :sweat_smile:. If all's good there, then feel free to merge this unless someone has a reason not to do it yet.

**Comment by markples** [comment]
Thank you for reviewing, launching the test, and merging!

**Pull request merged**